index.js

```diff
@@ -605,13 +605,29 @@ module.exports = class Hypercore extends EventEmitter {
     return this.core && this.core.globalCache
   }
 
+  get stats () {
+    return this.core !== null
+      ? this.core.stats
+      : {
+          blocksUploaded: 0,
+          blocksDownloaded: 0,
+          bytesUploaded: 0,
+          bytesDownloaded: 0,
+          blocksAppended: 0,
+          bytesAppended: 0
+        }
+  }
+
   ready () {
     return this.opening
   }
 
   _onupload (index, value, from) {
     const byteLength = value.byteLength - this.padding
 
+    this.core.stats.blocksUploaded++
+    this.core.stats.bytesUploaded += byteLength
```

Comment:
should not live in the sesison, move to replicator or who ever is triggering this

Reply:
This is emitted from the replicator, but the replicator has no direct access to the `byteLength` as calculated here, so I need to look into another way to calculate that (from what I can tell, it has no access to info about the encryption and its padding, but it can probably know the byte length of the block by checking Hypercore.core, which it does have access to).

Reply:
byteLength is value.byteLength, this is actualy wrong from a stats pov

Reply:
the padding is due to the session being encrypted, nothing to do with the size of the block it pulled from the network